README.md

```diff
@@ -11,572 +11,528 @@
-or a stream generator. Such generators can be combined together efficiently
-allowing the compiler to perform stream fusion and implement high performance
-stream merging combinators.
-
-## File IO
-
-The following code snippets implement some common Unix command line utilities
-using streamly.  You can compile these with `ghc -O2 -fspec-constr-recursive=16
--fmax-worker-args=16` and compare the performance with regular GNU coreutils
-available on your system.  Though many of these are not most optimal solutions
-to keep them short and elegant. Source file
-[HandleIO.hs](https://github.com/composewell/streamly/tree/master/examples/HandleIO.hs)
-in the examples directory includes these examples.
+## Reactive Programming
 
-``` haskell
-module Main where
-
-import qualified Streamly.Prelude as S
-import qualified Streamly.Data.Fold as FL
-import qualified Streamly.Data.Array.Foreign as A
-import qualified Streamly.FileSystem.Handle as FH
-import qualified System.IO as FH
-
-import Data.Char (ord)
-import System.Environment (getArgs)
-import System.IO (openFile, IOMode(..), stdout)
-
-withArg f = do
-    (name : _) <- getArgs
-    src <- openFile name ReadMode
-    f src
-
-withArg2 f = do
-    (sname : dname : _) <- getArgs
-    src <- openFile sname ReadMode
-    dst <- openFile dname WriteMode
-    f src dst
-```
+Streamly supports reactive and time domain programming inherently because of
+declarative concurrency. See the `Streamly.Prelude` module for some time
+specific combinators like `intervalsOf` and folds like `takeInterval` in
+`Streamly.Internal.Data.Fold`.  Also see pre-release sampling combinators in
+the `Streamly.Internal.Data.Stream.IsStream.Top` module including `throttle`
+and `debounce` like operations.
 
-### cat
+See
+[AcidRain.hs](https://github.com/composewell/streamly-examples/tree/master/examples/AcidRain.hs)
+and
+[CirclingSquare.hs](https://github.com/composewell/streamly-examples/tree/master/examples/CirclingSquare.hs).
 
-``` haskell
-cat = S.fold (FH.writeChunks stdout) . S.unfold FH.readChunks
-main = withArg cat
-```
+## More examples
 
-### cp
+Many more examples can be found in the [streamly-examples
+repository](https://github.com/composewell/streamly-examples).
 
-``` haskell
-cp src dst = S.fold (FH.writeChunks dst) $ S.unfold FH.readChunks src
-main = withArg2 cp
-```
+Streamly comes equipped with a very powerful set of abstractions to accomplish
+any kind of programming tasks that you may want to throw at it. It provides,
+streams, arrays, file-io, fsnotify, network-io, time domain programming
+(reactive programming). See the [streamly
+documentation](https://streamly.composewell.com) to know more.
 
-### wc -l
+## Concurrency
 
-``` haskell
-wcl = S.length . S.splitOn (== 10) FL.drain . S.unfold FH.read
-main = withArg wcl >>= print
-```
+Streamly uses lock-free synchronization for concurrent operations. It employs
+auto-scaling of the degree of concurrency based on demand. For CPU bound tasks
```

Comment:
>  It employs auto-scaling of the degree of concurrency based on demand.

Can this be rephrased, It's not easy to understand this.

Reply:
will do.